### PR TITLE
Added triggers page to Reference section

### PR DESCRIPTION
**Pull request by burtonr**
This adds a page to describe the various ways an OpenFaaS function can be triggered.
Currently only pointing to various repositories.
This will provide way for users to quickly and easily see the ways
a function can be invoked

Signed-off-by: Burton Rheutan <rheutan7@gmail.com>

### CODE REVIEW

**Comment by burtonr** [comment]
Here is the preview site for review: https://5b5091fefdd72a4963249f15--openfaas.netlify.com/reference/triggers/

File: docs/reference/triggers.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Triggers
+
+This is a list of the current options that are available to trigger an OpenFaaS Funtion
+
+## CLI
```
Comment (alexellis) [inline docs/reference/triggers.md]: I'd place the "CLI" section under the HTTP.

File: docs/reference/triggers.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Trigger a function using the `faas-cli` by using the function name
+
+```
+echo "triggered" | faas-cli invoke figlet
```
Comment (alexellis) [inline docs/reference/triggers.md]: CLI also support async mode.

File: docs/reference/triggers.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+This is the default, and standard method for interacting with your Functions.
+
+The function URL follows the pattern of:
+```
+https://<gateway URL>:<port>/function/<function name>
+```
+
+> There is also the ability to execute a function asynchronously by replacing `/function/` with `/async-function/` before the function name
+
+## Kafka
```
Comment (alexellis) [inline docs/reference/triggers.md]:
```
### Other event-sources

#### Kafka
```

File: docs/reference/triggers.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+Trigger a function from AWS SNS Notifications and Subscriptions
+
+More information in the repository: [affix/OpenFaaS-SNS](https://github.com/affix/OpenFaaS-SNS)
+
+## RabbitMQ
+Invoke functions from RabbitMQ topics
+
+More information in the repository: [Templum/rabbitmq-connector](https://github.com/Templum/rabbitmq-connector)
+
+## Azure EventGrid - CloudEvents
```
Comment (alexellis) [inline docs/reference/triggers.md]: Heading: CloudEvents

Within the body state it is an example for Azure EventGrid, but that any CloudEvent is supported in the function when sent over HTTP.

Comment (alexellis) [reply docs/reference/triggers.md]: Move this above Rabbit

File: docs/reference/triggers.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Triggers
+
+This is a list of the current options that are available to trigger an OpenFaaS Funtion
```
Comment (alexellis) [inline docs/reference/triggers.md]: OpenFaaS functions can be triggered easily by any kind of event. A small piece of code will convert from the event-source and trigger the function using the OpenFaaS Gateway API.

The most common use-case is HTTP which acts as a lingua franca between internet-connected systems.

File: docs/reference/triggers.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
```
Comment (alexellis) [inline docs/reference/triggers.md]: Link to the "Cron" page in the docs stating about period scheduling/triggers.

**Comment by burtonr** [comment]
Updated based on the feedback.

New preview site: https://deploy-preview-48--openfaas.netlify.com/reference/triggers/

**Pull request merged**